### PR TITLE
chore(repo): Improve E2E docs & QoL improvements

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

After going through some longer debugging sessions with our E2E tests I found room for QoL improvements and some more docs.

This PR adds instructions to the E2E README about debugging and more variants of running the tests. This PR also adds a `verbose` input to the `init` action to change the `TURBO_ARGS`.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: .github/actions/init/action.yml [hunk 1]
```diff
@@ -62,15 +67,24 @@ runs:
...
+
+          const ARGS = [
+            `--cache-dir=${CACHE_DIR}`,
+            `--concurrency=${cpus}`,
+            `--summarize=${SUMMARIZE}`,
+          ]
+
+          if (VERBOSE === 'true') {
+            ARGS.push('--verbosity=2')
+            ARGS.push('--log-order=stream')
```
Comment (LekoArts) [inline .github/actions/init/action.yml:80]: This ensures that logs are flushed instantly

File: .github/actions/init/action.yml [hunk 2]
```diff
@@ -62,15 +67,24 @@ runs:
...
+            ARGS.push('--output-logs=full')
```
Comment (LekoArts) [inline .github/actions/init/action.yml:81]: This is the default (right now) but setting it explicitly just to be safe for the future

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -170,10 +170,10 @@ jobs:
...
 
-      # - name: Upload Integration Report for ${{ matrix.test-name }}
-      #   uses: actions/upload-artifact@v3
-      #   if: always()
-      #   with:
-      #     name: integration-report-${{ github.run_id }}-${{ github.run_attempt }}-${{ matrix.test-name }}
-      #     path: playwright-report/
-      #     retention-days: 1
+      - name: Upload test-results
+        if: ${{ cancelled() || failure() }}
```
Comment (LekoArts) [inline .github/workflows/ci.yml:174]: We only record traces on failures so wouldn't make sense to run this when everything works

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -170,10 +170,10 @@ jobs:
...
+        uses: actions/upload-artifact@v4
+        with:
+          name: playwright-traces-${{ github.run_id }}-${{ github.run_attempt }}-${{ matrix.test-name }}
+          path: integration/test-results
```
Comment (LekoArts) [inline .github/workflows/ci.yml:178]: The default for [outputDir](https://playwright.dev/docs/api/class-testproject#test-project-output-dir) is `test-results`

File: integration/README.md [hunk 5]
```diff
@@ -510,10 +575,7 @@ The integration suite uses [`presets/envs.ts`](../integration/presets/envs.ts) t
 
 This is why you created the `.keys.json` file in the [initial setup](#initial-setup) step. Those secret and publishable keys are used to create environment configs. Inside GitHub actions these keys are provided through the `INTEGRATION_INSTANCE_KEYS` environment variable.
 
-Currently, we have two Clerk instances configured:
-
-- **with-email-codes**: a single session application with all toggles enabled.
-- **with-email-links**: a single session app with email links enabled and email codes disabled. Useful to test email links flows as the `<SignUp />` component currently does not support switching between email links and email codes.
+They keys defined in `.keys.json.sample` correspond with the Clerk instances in the **Integration testing** organization.
```
Comment (LekoArts) [inline integration/README.md:578]: This was already outdated, I don't think it's worth it to keep this in sync over time

File: integration/playwright.config.ts [hunk 6]
```diff
@@ -21,7 +21,7 @@ export const common: PlaywrightTestConfig = {
   workers: process.env.CI ? numAvailableWorkers : '70%',
   reporter: process.env.CI ? 'line' : 'list',
   use: {
-    trace: 'on-first-retry',
+    trace: 'retain-on-failure',
```
Comment (LekoArts) [inline integration/playwright.config.ts:24]: We might have flaky tests that require a retry, we're not really interested in traces for that

File: .github/actions/init/action.yml (hunk 1 above)
Comment (nikosdouvlis) [reply .github/actions/init/action.yml:80]: #til

File: integration/README.md (hunk 5 above)
Comment (nikosdouvlis) [reply integration/README.md:578]: Correct ✅  The keys in our shared secure notes (and in GH secrets of course) are always up to date so this was not needed!

**Review by nikosdouvlis — APPROVED** [review]
Nice!

**Pull request merged**